File: LeetCode/3021/python/lc3021_test.py
import unittest

from lc3021 import Solution

class Test_flowerGame(unittest.TestCase):
    def test(self):
        s = Solution()
        self.assertEqual(s.flowerGame(3, 2), 3)
        self.assertEqual(s.flowerGame(1, 1), 0)

if __name__ == "__main__":
    unittest.main()

File: LeetCode/3021/golang/lc3021_test.go
package lc3021

import "testing"

func Test_flowerGame(t *testing.T) {
	testcases := []struct {
		n        int
		m        int
		expected int64
	}{
		{3, 2, int64(3)},
		{1, 1, int64(0)},
	}

	for i, tc := range testcases {
		actual := flowerGame(tc.n, tc.m)
		if actual != tc.expected {
			t.Errorf("Testcase %02d (%v) failed: want %d, got %d", i+1, tc.n, tc.expected, actual)
		}
	}
}
